### PR TITLE
Fix state encryption regression + add integration test

### PR DESCRIPTION
**Pull request by yaron2**
Fixes https://github.com/dapr/dapr/issues/7512.

Adds basic integration test for state encryptions.

### CODE REVIEW

**Review by ItalyPaleAle — DISMISSED** [review]

File: tests/integration/suite/daprd/state/http/encryption.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2023 The Dapr Authors
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/state/http/encryption.go:2]: nit

```suggestion
Copyright 2024 The Dapr Authors
```

File: tests/integration/suite/daprd/state/http/encryption.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	procdaprd "github.com/dapr/dapr/tests/integration/framework/process/daprd"
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/state/http/encryption.go:30]: nit

```suggestion
	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
```

File: tests/integration/suite/daprd/state/http/encryption.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  version: v1
+  metadata:
+  - name: primaryEncryptionKey
+    secretKeyRef:
+      name: key
+auth:
+  secretStore: secretstore
+`
+
+	os.WriteFile(secretsFile, []byte(secretsJson), 0o600)
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/state/http/encryption.go:75]:
```suggestion
	require.NoError(t, os.WriteFile(secretsFile, []byte(secretsJson), 0o600))
```

File: tests/integration/suite/daprd/state/http/encryption.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+type encryption struct {
+	daprd *procdaprd.Daprd
+}
+
+func (e *encryption) Setup(t *testing.T) []framework.Option {
+	tmp := t.TempDir()
+	secretsFile := filepath.Join(tmp, "secrets.json")
+	secretsJson := `{ "key":  "5ca12365a9c1cc981090331f7eec683d"}` //nolint:gosec
```
Comment (JoshVanL) [inline tests/integration/suite/daprd/state/http/encryption.go:46]: Can we please generate this key at test runtime? Seems minor, but this kind of stuff does end up getting copied in places it shouldn't.

https://programmingwithstyle.com/posts/howihackedmycar/

**Review by JoshVanL — CHANGES_REQUESTED** [review]

File: tests/integration/suite/daprd/state/http/encryption.go (hunk 1 above)
Comment (yaron2) [reply tests/integration/suite/daprd/state/http/encryption.go]: Addressed

File: tests/integration/suite/daprd/state/http/encryption.go (hunk 3 above)
Comment (yaron2) [reply tests/integration/suite/daprd/state/http/encryption.go]: Addressed

File: tests/integration/suite/daprd/state/http/encryption.go (hunk 4 above)
Comment (yaron2) [reply tests/integration/suite/daprd/state/http/encryption.go]: Addressed

File: tests/integration/suite/daprd/state/http/encryption.go (hunk 2 above)
Comment (yaron2) [reply tests/integration/suite/daprd/state/http/encryption.go]: Addressed

**Review by artursouza — APPROVED** [review]

**Pull request merged**